### PR TITLE
Fix SP precision builds

### LINKED ISSUE #119: Compiling against single precision OpenFOAM leads to undefined symbol errors
Currently, we only support DP scalars. Compiling against SP OpenFOAM leads to the following error.

```
v2312/platforms/linux64GccSPInt32Opt/lib/libOGL.so: undefined symbol: _ZNK4Foam9lduMatrix6solver11scalarSolveERNS_5FieldIdEERKS3_h
```

A workaround is to modify the CMakeLists.txt and set

```
  target_compile_definitions(OGL PUBLIC WM_LABEL_SIZE=32 WM_ARCH_OPTION=32
              NoRepository WM_SP)
```

here the important bit is to set `WM_SP` since this will be used with #ifdef macros to set the scalar type in scalar.H. Eventually this should be done automatically by cmake, by reading` $WM_LABEL_SIZE ` and` $WM_PRECISION_OPTION`.

### PR DESCRIPTION
**Pull request by greole**
This PR improves build for SP precission.
- We now read the label and scalar size from the environment variables. This makes `OGL_DP_LABELS` obsolete.  This PR fixes #119. 

Additional changes:
- If building with `-DOGL_USE_EXTERNAL_GINKGO=ON` no backend needs to be specified
- Add full_ci label for integration tests on AWS.

### CODE REVIEW

File: .github/workflows/compile_on_aws.yml [hunk 1]
```diff
@@ -4,11 +4,18 @@
...
 on:
-  # enable to manually trigger the tests
-  workflow_dispatch:
+  pull_request:
+    types: synchronize
+      # workflow_run:
+      #   workflows: ["Integration Tests"]
+      #   types:
+      #     - completed
+
```
Comment (greole) [inline .github/workflows/compile_on_aws.yml]:
```suggestion

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ SPDX-License-Identifier: GPL-3.0-or-later
 -->
 
 # 0.5.4 (unreleased)
+- Add support for SP scalars and determine label and scalar size from env variable [PR #120](https://github.com/hpsim/OGL/pull/120) 
+- Notify user of a unsupported executor argument [PR #118](https://github.com/hpsim/OGL/pull/118) 
 - Notify user of a unsupported executor argument [PR #]
 - Add a list of all OGL authors [PR #113](https://github.com/hpsim/OGL/pull/113)
```
Comment (chihtaw) [inline CHANGELOG.md]: Should be: Add a list of all OGL authors [PR #126](https://github.com/hpsim/OGL/pull/126)

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,8 @@ SPDX-License-Identifier: GPL-3.0-or-later
...
 - Replace #ifdef include guard with #pragma once [PR #113](https://github.com/hpsim/OGL/pull/113)
```
Comment (chihtaw) [inline CHANGELOG.md]: Replace #ifdef include guard with #pragma once [PR #126](https://github.com/hpsim/OGL/pull/126)

File: CHANGELOG.md (hunk 2 above)
Comment (greole) [inline CHANGELOG.md]:
```suggestion
- Add a list of all OGL authors [PR #126](https://github.com/hpsim/OGL/pull/126)
```

Comment (greole) [reply CHANGELOG.md]: But in generall I think this not even needed to add here, since it is not   relevant for users.

File: CHANGELOG.md (hunk 3 above)
Comment (greole) [inline CHANGELOG.md]:
```suggestion
- Replace #ifdef include guard with #pragma once [PR #126](https://github.com/hpsim/OGL/pull/126)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,9 +5,11 @@ SPDX-License-Identifier: GPL-3.0-or-later
...
 
 # 0.5.4 (unreleased)
+- Add support for SP scalars and determine label and scalar size from env variable [PR #120](https://github.com/hpsim/OGL/pull/120) 
+- Notify user of a unsupported executor argument [PR #118](https://github.com/hpsim/OGL/pull/118) 
 - Notify user of a unsupported executor argument [PR #]
-- Add a list of all OGL authors [PR #113](https://github.com/hpsim/OGL/pull/113)
-- Replace #ifdef include guard with #pragma once [PR #113](https://github.com/hpsim/OGL/pull/113)
+- Add a list of all OGL authors [PR #126](https://github.com/hpsim/OGL/pull/126)
+- Replace #ifdef include guard with #pragma once [PR #126](https://github.com/hpsim/OGL/pull/126)
 - Switch to SPDX headers, fix formatting issue[PR #112](https://github.com/hpsim/OGL/pull/112)
```
Comment (chihtaw) [inline CHANGELOG.md]: Switch to SPDX headers, fix formatting issue[PR #122](https://github.com/hpsim/OGL/pull/122)

Comment (greole) [inline CHANGELOG.md]:
```suggestion
- Switch to SPDX headers, fix formatting issue[PR #122](https://github.com/hpsim/OGL/pull/112)
```

Comment (greole) [inline CHANGELOG.md]: fix
```suggestion
- Switch to SPDX headers, fix formatting issue[PR #122](https://github.com/hpsim/OGL/pull/112)
```

**Pull request merged**